test/lib/generateToken.test.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const assert = require('chai').assert;
+const mockery = require('mockery');
+const sinon = require('sinon');
+const crypto = require('crypto');
+
+sinon.assert.expose(assert, { prefix: '' });
+require('sinon-as-promised');
+
+describe('generateToken', () => {
+    const randomBytes = 'some random bytes';
+    // Result of passing 'some random bytes' through a sha256 hash, in base64
+    const hash = 'mF0EvjvxWMrVz5ZGJcnbe0ZPooUlv/DAB9VrV6bmZmg=';
+    let nodeifyMock;
+    let generateToken;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+        nodeifyMock = {
+            withContext: sinon.stub()
+        };
+
+        mockery.registerMock('./nodeify', nodeifyMock);
```

Comment:
nodeify is a runtime dependency that shouldn't be mocked. You can spy on it if you want to.